### PR TITLE
Refactor configuration classes

### PR DESCRIPTION
**Pull request by timja**
I've done the following:
* Reduced visibility in the CLI, public fields break encapsulation and the `CLIOption` class didn't need to be visible at all.
* Introduced a builder to the `Config` class to make constructing with defaults simple, while having an immutable `Config` class, and not having to maintain multiple constructors when new fields are added
* Removed `ConfigSetup` class, not really needed in my opinion, although now the CLIOption class is doing more than just reading in values
* Converted update center getters to `URL`
* Converted a for loop to a stream
* Removed pluginTxt field from config, it wasn't being used there at all and the library shouldn't have known about it, its only concerned about plugins
* Added missing field for showAllWarnings in Config

cc @stopalopa @kwhetstone

### CODE REVIEW

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/config/Config.java [hunk 1]
```diff
@@ -2,94 +2,156 @@
...
+ * ...
+ * build()
+ * }
+ * <br/>
+ * Defaults for update centers will be set for you
+ */
 public class Config {
     private File pluginDir;
-    private File pluginTxt;
     private boolean showWarnings;
```
Comment (timja) [inline plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/config/Config.java:23]: @stopalopa this field doesn't seem to be in use, is it for later or a mistake? the getter is never called.

File: plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java [hunk 2]
```diff
@@ -1,104 +1,272 @@
...
     @Option(name = "--plugin-file", aliases = {"-f"}, usage = "Path to plugins.txt file",
             handler = FileOptionHandler.class)
-    public File pluginTxt;
+    private File pluginTxt;
 
     @Option(name = "--plugin-download-directory", aliases = {"-d"},
             usage = "Path to directory in which to install plugins",
             handler = FileOptionHandler.class)
-    public File pluginDir;
+    private File pluginDir;
```
Comment (stopalopa) [inline plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java:34]: So I had kind of modeled what I was doing after the custom war packager. Is there a reason why in the CLI for that, all of the CLI variables are public? https://github.com/jenkinsci/custom-war-packager/blob/master/custom-war-packager-cli/src/main/java/io/jenkins/tools/warpackager/cli/CliOptions.java

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/config/Config.java (hunk 1 above)
Comment (stopalopa) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/config/Config.java:23]: I think it's fine to remove.

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/config/Config.java [hunk 3]
```diff
@@ -2,94 +2,156 @@
...
+    public static class Builder {
+        private File pluginDir;
+        private boolean showWarnings;
+        private boolean showAllWarnings;
+        private String jenkinsWar;
+        private List<Plugin> plugins = new ArrayList<>();
+        ;
+        private URL jenkinsUc = Settings.DEFAULT_UPDATE_CENTER;
+        private URL jenkinsUcExperimental = Settings.DEFAULT_EXPERIMENTAL_UPDATE_CENTER;
+        private URL jenkinsIncrementalsRepoMirror = Settings.DEFAULT_INCREMENTALS_REPO_MIRROR;
```
Comment (stopalopa) [inline plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/config/Config.java]: I think there needs to be some type of handling for malformed exceptions.

**Pull request merged**